src/v/utils/uuid.h

```diff
@@ -0,0 +1,74 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#pragma once
+
+#include "bytes/details/out_of_range.h"
+#include "reflection/adl.h"
+
+#include <absl/hash/hash.h>
+#include <boost/functional/hash.hpp>
+#include <boost/uuid/uuid.hpp>
+#include <boost/uuid/uuid_io.hpp>
+
+#include <vector>
+
+// Wrapper around Boost's UUID type suitable for serialization with serde.
+// Provides utilities to construct and convert to other types.
+//
+// Expected usage is to supply a UUID v4 (i.e. based on random numbers). As
+// such, serialization of this type simply serializes 16 bytes.
+struct uuid_t {
+public:
+    static constexpr auto length = 16;
+    using underlying_t = boost::uuids::uuid;
+
+    underlying_t uuid;
+
+    explicit uuid_t(const underlying_t& uuid)
+      : uuid(uuid) {}
+
+    explicit uuid_t(const std::vector<uint8_t>& v)
+      : uuid({}) {
+        if (v.size() != length) {
+            details::throw_out_of_range(
+              "Expected size of {} for UUID, got {}", length, v.size());
+        }
+        std::copy(v.begin(), v.end(), uuid.begin());
+    }
+
+    uuid_t() noexcept = default;
+
+    std::vector<uint8_t> to_vector() const {
```

Comment:
if there currently is no consumer of this method, i'd just leave it out

Reply:
Yeah this should be a part of a different PR I guess, or added in a test. I will use this soon.